src/Avalonia.Base/Input/InputElement.cs

```diff
@@ -504,6 +504,21 @@ public bool Focus(NavigationMethod method = NavigationMethod.Unspecified, KeyMod
             return FocusManager.GetFocusManager(this)?.Focus(this, method, keyModifiers) ?? false; 
         }
 
+        /// <summary>
+        /// Cancels the current gesture recognizer captured by the pointer, and restores the previous captured 
+        /// element if available. If no gesture is currently started by the pointer, no changes are made.
+        /// </summary>
+        /// <param name="pointer">The pointer with the captured gesture</param>
+        public void CancelCurrentGesture(Pointer pointer)
```

Comment:
The API usage is unclear. Which element one would call this method on?

Reply:
i. e. 
```xml
<ScrollViewer> <!-- captured here -->
   <StackPanel>
       <TextBox .../>  <-- I assume it's the control that would want to cancel the gesture -->
   </StackPanel>
</ScrollViewer>
```

Reply:
ScrollViewer, in this case.

Reply:
Do we need to know that we actually canceled some active gesture?

I could call that at the root level and the method would walk down the visual tree to find a matching control. Do you think this is really needed or should the caller implement this?